napari/_qt/qt_viewer.py

```diff
@@ -172,24 +153,33 @@ class QtViewer(QSplitter):
         QWidget wrapped in a QDockWidget with forwarded viewer events.
     dockLayerList : QtViewerDockWidget
         QWidget wrapped in a QDockWidget with forwarded viewer events.
+    key_map_handler : napari.utils.KeymapHandler
+        KeymapHandler handling the calling functionality when keys are pressed that have a callback function mapped.
     layerButtons : QtLayerButtons
         Button controls for napari layers.
     layers : QtLayerList
         Qt view for LayerList controls.
-    layer_to_visual : dict
-        Dictionary mapping napari layers with their corresponding vispy_layers.
-    view : vispy scene widget
-        View displayed by vispy canvas. Adds a vispy ViewBox as a child widget.
+    qt_poll : Optional[QtPoll]
+        A QtPoll object required for octree or monitor.
+    remote_manager : RemoteManager
+        A remote manager processing commands from remote clients and sending out messages when polled.
+    show_welcome_screen : bool
+        Boolean indicating whether to show the welcome screen.
     viewer : napari.components.ViewerModel
         Napari viewer containing the rendered scene, layers, and controls.
     viewerButtons : QtViewerButtons
         Button controls for the napari viewer.
+    welcome_widget : QtWidgetOverlay
+        QtWidgetOverlay providing the stacked widgets for the welcome page.
     """
 
     _instances = WeakSet()
 
     def __init__(
-        self, viewer: ViewerModel, show_welcome_screen: bool = False
+        self,
+        viewer: ViewerModel,
+        show_welcome_screen: bool = False,
+        canvas_class: Type[VispyCanvas] = VispyCanvas,
```

Comment:
Is this parameter specified anywhere internally within napari? I could not find any such usage.

Or is this only for external usage? If so, I think I'm against that, but could be convinced with a good reason.

Otherwise, I think we should just remove this parameter.

Reply:
This is implemented based on the comment of @Czaki during one of the last community meetings. From the agenda:
- Grzegorz: would be good to be able to configure the VisPy Canvas class used by the viewer. e.g. pass argument `vispy_canvas_class=VispyCanvas`, then instantiate with `vispy_canvas_class()`. Ex: [`actual_factory` in `make_napari_viewer` fixture](https://github.com/napari/napari/blob/ff1e56ef8e44fa0636d6295e5a001e443624b5e7/napari/utils/_testsupport.py#L221).

Reply:
This is for situations when Embedding QtViewer in an application to keep the possibility of overwriting methods moved to canvas.

Reply:
OK, I'm convinced!